tsdb/db_test.go

```diff
@@ -7379,3 +7388,108 @@ func TestGenerateCompactionDelay(t *testing.T) {
 		assertDelay(db.generateCompactionDelay())
 	}
 }
+
+type blockedResponseRecorder struct {
+	r *httptest.ResponseRecorder
+
+	writeUnblocked chan struct{}
+	writeStarted   chan struct{}
+}
+
+func (br *blockedResponseRecorder) Write(buf []byte) (int, error) {
+	select {
+	case <-br.writeStarted:
```

Comment:
I added some comments, hope it's clearer now.

Reply:
@machine424 I still think this comment would be helpful 😅

Reply:
> It would also help me if you hadn't force pushed, because the force push means I can't see what changed since last time. Reading the PR again.

Sorry for that, the problem could also be that I rebased on main and the PR was opened for quite some time.

> @machine424 I still think this comment would be helpful 😅 Since what wasn't clear to me, initially, was the mechanics of the select itself.

I usually avoid transcribing what the code does in comments (avoid divergence, keep it readable), one can always `ctrl+F` to see when it receives a value (in this case closed.)